### PR TITLE
test: add vulncheck to nightly builds

### PR DESCRIPTION
**Pull request by mmulji-ic**
## Description

Partially addresses: #1879 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules) --> NA
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc) --> NA
* [ ] Updated the relevant documentation or specification --> will be addressed in another task
* [x] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: .github/workflows/nightly-tests.yml [hunk 1]
```diff
@@ -13,4 +23,49 @@ jobs:
...
 
-    
+  run-vulncheck:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3.3.0
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.1
```
Comment (mmulji-ic) [inline .github/workflows/nightly-tests.yml]: // comment
Govulncheck does not work on go 1.18.x and does not target a specific go version.
We should run govulncheck with the same go version as we intend to build Gaia.

Comment (mpoke) [reply .github/workflows/nightly-tests.yml]: Gaia on main is built with go 1.20 https://github.com/cosmos/gaia/blob/c26737455aa306a6e1c285b8860fc26898dcb1b4/Makefile#L24

Comment (sainoe) [reply .github/workflows/nightly-tests.yml]:
```suggestion
          go-version: 1.20.x
```

Comment (mmulji-ic) [reply .github/workflows/nightly-tests.yml]: Updated task to run against go v.1.20.3

**Review by mpoke — APPROVED** [review]
LGTM

**Review by sainoe — APPROVED** [review]

**Pull request merged**